pkg/docker/config/config_test.go

```diff
@@ -675,3 +676,170 @@ func TestAuthKeysForRef(t *testing.T) {
 		require.Equal(t, tc.expected, result, tc.name)
 	}
 }
+
+func TestSetCredentials(t *testing.T) {
+	for _, tc := range []struct {
+		registries []string
+		assert     func(dockerConfigFile)
+	}{
+		{
+			registries: []string{"quay.io"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 1)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+			},
+		},
+		{
+			registries: []string{"quay.io/a/b/c/d/image"},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 1)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c/d/image"].Auth)
+			},
+		},
+		{
+			registries: []string{
+				"quay.io/a/b/c",
+				"quay.io/a/b",
+				"quay.io/a",
+				"quay.io",
+				"my-registry.local",
+				"my-registry.local",
+			},
+			assert: func(auth dockerConfigFile) {
+				require.Len(t, auth.AuthConfigs, 5)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b/c"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a/b"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io/a"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["quay.io"].Auth)
+				require.NotEmpty(t, auth.AuthConfigs["my-registry.local"].Auth)
+			},
+		},
+	} {
+		tmpFile, err := ioutil.TempFile("", "auth.json.set")
+		require.NoError(t, err)
+		_, err = tmpFile.WriteString("{}")
+		require.NoError(t, err)
+		defer os.RemoveAll(tmpFile.Name())
+		sys := &types.SystemContext{AuthFilePath: tmpFile.Name()}
+
+		for _, registry := range tc.registries {
```

Comment:
The variable is not a registry (== host[:port]); the array is not `registries`.

Reply:
Yes, I changed it to `input` to be more generic.